src/bot/handlers.py

```diff
@@ -78,13 +84,14 @@ async def select_subcategory_callback(update: Update, context: ContextTypes.DEFA
         return
 
     subcategory_id = int(subcategory_match.group(1))
-    selected_categories = context.user_data.get("selected_categories", {})
-    if subcategory_id in selected_categories:
+
+    selected_categories = context.user_data.setdefault("selected_categories", {})
+
+    if subcategory_id in selected_categories.keys():

```

Comment:
```suggestion
    if subcategory_id in selected_categories:

```
Так будет работать быстрее.